azurerm/data_source_public_ip_prefix.go

```diff
@@ -0,0 +1,65 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmPublicIpPrefix() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmPublicIpPrefixRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"location": azure.SchemaLocationForDataSource(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"sku": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"prefix_length": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"ip_prefix": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"zones": azure.SchemaSingleZone(),
+
+			"tags": tags.Schema(),
```

Comment:
since this is a Data Source these are ReadOnly - as such could we make this:

```suggestion
			"tags": tags.SchemaDataSource(),
```

Reply:
Thanks for finding this.

I was curious what the difference is?
It [looks like](https://gist.github.com/midacts/17125d9b0dd669d973218fddea1ebd60/revisions) `SchemaDataSource` removes these:

- `Optional: true`
- `ValidateFunc: Validate` 
  - I see that this calls `Validate` to validate the tags

and adds:

```text
Elem: &schema.Schema{
	Type: schema.TypeString,
},
```

I am not sure what the `Elem` part does.